File: go/minimum-time-to-make-rope-colorful.go
package main

func minCost(colors string, neededTime []int) int {
  totalTime := 0
  cursor := 0

  for i := 1; i < len(colors); i++ {
    if colors[i] == colors[cursor] {
      if neededTime[i] < neededTime[cursor] {
        totalTime += neededTime[i]
        continue
      }
      totalTime += neededTime[cursor]
    }
    cursor = i
  }
  return totalTime
}

File: js/minimum-time-to-make-rope-colorful.js
export const minCost = (colors, neededTime) => {
  let totalTime = 0;
  let cursor = 0;

  for (let i = 1; i < colors.length; i++) {
    if (colors[i] === colors[cursor]) {
      if (neededTime[i] < neededTime[cursor]) {
        totalTime += neededTime[i];
        continue;
      }
      totalTime += neededTime[cursor];
    }
    cursor = i;
  }

  return totalTime;
};
